src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IClientCapabilitiesProvider.cs

```diff
@@ -4,10 +4,13 @@
 
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 
-namespace Microsoft.CodeAnalysis.LanguageServer
+namespace CommonLanguageServerProtocol.Framework;
+
+public interface IClientCapabilitiesProvider
 {
-    internal interface ICapabilitiesProvider
-    {
-        ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities);
-    }
+    ClientCapabilities GetClientCapabilities();
+
+    bool HasBeenSet();
```

Comment:
That would be because I've just created it :) The idea here was that there were a couple places where people were creating full clientServer objects just to get the ClientCapabilities off of it. I thought it might be better if we de-coupled to two to keep such a big object from being needlessly included.

Reply:
I see I see. Do handlers need to ever acquire this service given the request execution queue has the client capabilities passed into it?

Reply:
The SemanticTokensRangeHandler for Roslyn currently needs this into when it's constructed, but after considering your feedback I re-structured it to do that work on first-request (which will have the ClientCapabilities).